src/map.cpp

```diff
@@ -1444,11 +1444,7 @@ MapSector *ServerMap::createSector(v2s16 p2d)
 	/*
 		Do not create over max mapgen limit
 	*/
-	const s16 max_limit_bp = MAX_MAP_GENERATION_LIMIT / MAP_BLOCKSIZE;
-	if (p2d.X < -max_limit_bp ||
-			p2d.X >  max_limit_bp ||
-			p2d.Y < -max_limit_bp ||
-			p2d.Y >  max_limit_bp)
+	if (blockpos_over_max_limit(v3s16(p2d.X, 0, p2d.Y)))
```

Comment:
What exactly is the purpose of this change? It looks like it fixes a bug, but it is unclear to me what the problem was here before and why this is part of this PR.

The code reuses blockpos_over_max_limit() with the Y component of a v3s16 set to 0 and the Z component of the v3s16 set to the Y component of a v2s16. Is this intentional?

Reply:
Just less code for the same thing, no behavioral change.

> Y component of a v3s16 set to 0 and the Z component of the v3s16 set to the Y component of a v2s16. Is this intentional?

Yes. The X and Z components of a block positions specify the sector it is in.